packages/@uppy/xhr-upload/src/index.js

```diff
@@ -90,7 +96,13 @@ module.exports = class XHRUpload extends Plugin {
        * @param {XMLHttpRequest | respObj} response the response object (XHR or similar)
        */
       getResponseError (responseText, response) {
-        return new Error('Upload error')
+        let error = new Error('Upload error')
+
+        if (isNetworkError(response)) {
+          error = new NetworkError(error)
```

Comment:
should we be passing the `xhr` object to the `NetworkError` here as well?

Reply:
Yep.